ksql-cli/src/test/java/io/confluent/ksql/TestRunner.java

```diff
@@ -51,8 +55,16 @@ protected static void test(String command, TestResult.OrderedResult expectedResu
   }
 
   protected static void test(String command, TestResult expectedResult, boolean requireOrder) {
-    TestResult actual = run(command, requireOrder);
-    Assert.assertEquals(expectedResult, actual);
+    run(command, requireOrder);
+    try {
+      TestUtils.waitForCondition(() -> {
+        TestResult actualResult = testTerminal.getTestResult();
+        return actualResult.data.containsAll(expectedResult.data);
+      }, 1000, "Did not get the expected result '" + expectedResult.toString() + ", in a timely fashion.");
```

Comment:
Maybe we should make the timeout a bit larger, i.e, 10 seconds? so that it is more reliable?